File: DTO/ScheduleItem.cs
using System.Text.Json.Serialization;

namespace Dishora.DTO
{
    public class ScheduleItem
    {
        [JsonPropertyName("scheduleId")]
        public long ScheduleId { get; set; }

        [JsonPropertyName("availableDate")]
        public string AvailableDate { get; set; } // "yyyy-MM-dd"

        [JsonPropertyName("maxOrders")]
        public int MaxOrders { get; set; }

        [JsonPropertyName("currentOrderCount")]
        public int CurrentOrderCount { get; set; } // Your Java class uses 'currentOrders'

        [JsonPropertyName("isActive")]
        public bool IsActive { get; set; }

        [JsonPropertyName("businessId")]
        public long BusinessId { get; set; }
    }
}

File: app/src/main/java/com/example/dishora/vendorUI/homeTab/schedule/model/ScheduleItem.java
package com.example.dishora.vendorUI.homeTab.schedule.model;

import com.google.gson.annotations.SerializedName;

import java.io.Serializable;

// Represents one schedule entry received from the ASP.NET Core API
public class ScheduleItem implements Serializable {

    @SerializedName("scheduleId")
    public long scheduleId;

    @SerializedName("availableDate")
    public String availableDate; // Matches C# string format "yyyy-MM-dd"

    @SerializedName("maxOrders")
    public int maxOrders;

    @SerializedName("currentOrderCount")
    public int currentOrders;

    @SerializedName("isActive")
    public boolean isActive;

    @SerializedName("businessId")
    public long businessId;

    // Optional: Keep the constructor simple for easy instantiation if needed,
    // but Gson uses reflection, so it doesn't strictly require this constructor
    public ScheduleItem(String availableDate, int maxOrders, int currentOrders) {
        this.availableDate = availableDate;
        this.maxOrders = maxOrders;
        this.currentOrders = currentOrders;
    }

    // Calculated method used by the RecyclerView Adapter
    public int getRemainingCapacity() {
        return maxOrders - currentOrders;
    }

    // You can also add getters for a cleaner coding style, but it's optional
    public String getAvailableDate() { return availableDate; }

    public boolean isFull() {
        // A date is considered "full" or unavailable if it's inactive
        // OR if the current orders meet or exceed the max capacity.
        if (!isActive) {
            return true;
        }
        return currentOrders >= maxOrders;
    }
}